src/main/java/com/example/MyBatisDemo/dto/SubjectAdditionDto.java

```diff
@@ -1,11 +1,17 @@
 package com.example.MyBatisDemo.dto;
 
+import com.example.MyBatisDemo.model.TeacherEntity;
+import jakarta.validation.constraints.NotBlank;
 import lombok.Data;
 
 @Data
 public class SubjectAdditionDto {
+    @NotBlank(message = "id is mandatory")
     private Long id;
+    @NotBlank(message = "name is mandatory")
     private String name;
+    @NotBlank(message = "credits is mandatory")
     private int credits;
-    private Long teacher;
+    @NotBlank(message = "teacher is mandatory")
+    private TeacherEntity teacher;
```

Comment:
Shouldn't we just accept teacherId over here as we are not creating a teacher entity?

Reply:
Yes updated now we are accepting only teacher Id